### DISCUSSION TITLE
How can i compile Vita2hos myself?

### DISCUSSION THREAD

**Discussion by lanana421** [xerpi/vita2hos#19] [Q&A] [answered]
Hello, i was wondering how i can compile this myself, Devkitpro watns libnx32, which i can't find anything about online.

**Comment by TSRBerry** [comment, marked as answer]
These projects are in the process of getting merged, so they are currently not publicly available.
Iirc as soon as libnx supports 32-bit, all dependencies should be public that are needed to build vita2hos yourself.

**Reply by blackjackfruit** [reply]
Is there a timeframe when to expect libnx32? I was hoping to build/play around with the vita2hos source to test on my switch, but I guess that isn't possible. Is there a work around to this requirement?

**Reply by TSRBerry** [reply]
Sadly not, as far as I know xerpi is still waiting on a response from dkp and switchbrew devs.

The only workaround would be to create a patch for libnx to support 32-bit yourself. :/
Other than that, Is there anything specific you want to do? I could ask xerpi about giving you access to our dev environment depending on your usecase.

**Reply by blackjackfruit** [reply]
This was more educational for me to understand ARM a bit better. I have some knowledge in C and ASM and wanted to see how much I understand the codebase.

**Reply by TSRBerry** [reply]
Awesome! I'll give you access :D
